trace-sdk/src/main/java/io/bitrise/trace/network/MetricSender.java

```diff
@@ -129,11 +130,8 @@ static int countHeaderComparisonDifference(@NonNull final MetricRequest request,
   public void onSuccess() {
     TraceLog.i(LogMessageConstants.METRIC_SENT_SUCCESSFULLY);
     getExecutor().execute(() -> {
-      if (!getMetricEntityList().isEmpty()) {
```

Comment:
i don't think this can actually happen - we check when we send if we have metrics to send - i don't think it would be changed before the request has been completed?